kafka-connect-s3/src/test/java/io/confluent/connect/s3/TopicPartitionWriterTest.java

```diff
@@ -650,6 +650,40 @@ public void testNoFilesWrittenWithoutCommit() throws Exception {
     verify(Collections.<String>emptyList(), -1, schema, records);
   }
 
+  @Test
+  public void testRotateIntervalIsIgnoredWhenUsedWithNoTimeBasedPartitioner() throws Exception {
+    // Setting size-based rollup to 10 but will produce fewer records. Commit should not happen.
+    localProps.put(S3SinkConnectorConfig.FLUSH_SIZE_CONFIG, "10");
+    localProps.put(
+        S3SinkConnectorConfig.ROTATE_INTERVAL_MS_CONFIG,
+        String.valueOf(TimeUnit.MINUTES.toMillis(1))
+    );
+    setUp();
+
+    // Define the partitioner
+    Partitioner<FieldSchema> partitioner = new DefaultPartitioner<>();
+    partitioner.configure(parsedConfig);
+    TopicPartitionWriter topicPartitionWriter = new TopicPartitionWriter(
+        TOPIC_PARTITION, storage, writerProvider, partitioner,  connectorConfig, context);
+
```

Comment:
this block of code is repeated a bunch of times. should we put it in a `@Before` method? And have `topicPartitionWriter` as a field in this class?

Reply:
Not sure which block are you referring to, but everything below `setUp` varies more or less between test cases (the partitioner or the records constructed). More importantly though I'd like to keep this fix as focused as possible because it goes all the way back to `3.2.x`

Thanks for the reviews!

Reply:
I meant this block:

```
    // Define the partitioner
    Partitioner<FieldSchema> partitioner = new DefaultPartitioner<>();
    partitioner.configure(parsedConfig);
    TopicPartitionWriter topicPartitionWriter = new TopicPartitionWriter(
        TOPIC_PARTITION, storage, writerProvider, partitioner,  connectorConfig, context);
```

I did a quick check, and this block is duplicated in 3.2.x as well.

Reply:
I was referring to possible conflicts that will be generated. But the partitioner is not the same across test cases.